src/shadowbox/server/server_access_key.ts

```diff
@@ -199,14 +207,22 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     const metricsId = uuidv4();
     const password = params?.password ?? generatePassword();
     const encryptionMethod = params?.encryptionMethod || this.NEW_USER_ENCRYPTION_METHOD;
+    const portNumber = params?.portNumber ?? this.portForNewAccessKeys;
 
-    // Validate encryption method.
+    // Validate encryption method and port number.
     if (!isValidCipher(encryptionMethod)) {
       throw new errors.InvalidCipher(encryptionMethod);
     }
+    if (!isValidPort(portNumber)) {
+      throw new errors.InvalidPortNumber(portNumber);
+    }
+    if (!(await this.isPortAvailable(portNumber))) {
```

Comment:
I'm not completely convinced we should do this check here, as it restricts concurrency. Today we don't check the availability of the port when it's being used (only when it's set in `setPortForNewAccessKeys`. Thoughts?

Reply:
Yeah, I think my point was that once you set a port, we never check it again, or do we? Anyway yeah I agree an if-block is better, will change that.